src/OpenTelemetry.Exporter.Geneva/TLDExporter/TLDTraceExporter.cs

```diff
@@ -0,0 +1,309 @@
+// <copyright file="TLDTraceExporter.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Exporter.Geneva.TLDExporter;
+
+internal sealed class TLDTraceExporter : TLDExporter, IDisposable
+{
+    private readonly string partAName = "Span";
+    private readonly byte partAFieldsCount = 3; // At least three fields: time, ext_dt_traceId, ext_dt_spanId
+    private readonly IReadOnlyDictionary<string, object> m_customFields;
+    private readonly Tuple<byte[], byte[]> repeatedPartAFields;
+    private static readonly string INVALID_SPAN_ID = default(ActivitySpanId).ToHexString();
+
+    private readonly EventProvider eventProvider;
+    private static readonly ThreadLocal<EventBuilder> eventBuilder = new(() => new(UncheckedASCIIEncoding.SharedInstance));
```

Comment:
Which one would be better?
1. Having 3 ThreadLocals
2. Having a single ThreadLocal which points to a complex struct/object with 3 fields

Reply:
I have no strong preference. I suppose there's no performance implications of using one over the other. If the second approach looks more readable, I could make that change.

Reply:
Added a TODO for this.